src/labels.js

```diff
@@ -313,9 +175,16 @@ class Labels {
     addr.label = label;
     addr.used = false;
 
-    this._walletNeedsSync = true;
+    // Update wallet:
+    let labels = this._wallet.hdwallet.accounts[accountIndex]._address_labels;
+
+    let labelEntry = {
+      index: receiveIndex,
+      label: label
+    };
+    labels.push(labelEntry);
 
-    return this.save().then(() => {
+    return this._syncWallet().then(() => {
```

Comment:
@jtormey I double checked that this `.then()` function indeed gets called. I also made `_syncWallet()` private to `Labels`.